File: src/main/java/sortingalgorithms/selectionsort/java/SelectionSort.java
package sortingalgorithms.selectionsort.java;

import org.jetbrains.annotations.NotNull;
import sortingalgorithms.SortingAlgorithm;

import java.util.Arrays;

public class SelectionSort extends SortingAlgorithm {
    @Override
    public void sort(@NotNull Integer[] input) {
        int length = input.length;
        for (int i = 0; i < length; i++) {
            int minIndex = i;
            for (int j = i; j < length; j++) {
                if (input[j] < input[minIndex]) {
                    minIndex = j;
                }
            }
            swap(input, minIndex, i);
        }
        Arrays.toString(input);
    }
}

File: src/main/java/sortingalgorithms/selectionsort/kotlin/SelectionSort.kt
package sortingalgorithms.selectionsort.kotlin

import sortingalgorithms.SortingAlgorithm

class SelectionSort : SortingAlgorithm() {
    override fun sort(input: IntArray) {
        val length = input.size
        for (i in 0 until length) {
            var minIndex = i
            for (j in i until length) {
                if (input[j] < input[minIndex]) {
                    minIndex = j
                }
            }
            swap(input, minIndex, i)
        }
    }
}